Automattic/wp-calypso
# Jetpack Cloud Landing: Use feature check for search

#### Changes proposed in this Pull Request

* Replaced uses of `siteHasSearchProductPurchase` with feature checks.
* Removes `siteHasSearchProductPurchase` selector.

#### Testing instructions



* Create a new site on [Jurassic Ninja](https://jurassic.ninja/specialops/).
* Purchase Jetpack search for it.
* Navigate to https://container-infallible-driscoll.calypso.live/landing and select the site you created.
* You should end up in https://container-infallible-driscoll.calypso.live/jetpack-search/SITE.



Related to p4TIVU-a66-p2
Depends on https://github.com/Automattic/wp-calypso/pull/63735
